pheps/phep-0003.md

```diff
@@ -0,0 +1,152 @@
+```
+PHEP: 3
+Title: PyHC Python Support Policy
+Author: Shawn Polson <pyhc-confidential@lasp.colorado.edu> <https://orcid.org/0000-0003-0619-5745>
+Discussions-To: https://github.com/heliophysicsPy/standards/pull/29
+Revision: 1
+Status: Draft
+Type: Standards Track
+Content-Type: text/markdown; charset=UTF-8; variant=CommonMark
+Created: 06-Jun-2024
+Post-History: 06-Jun-2024
+```
+
+# Abstract
+<a name="abstract"></a>
+This PHEP recommends that all projects across the PyHC ecosystem adopt a common time-based policy for support of Python versions, similar to [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html) and [SPEC-0](https://scientific-python.org/specs/spec-0000/). Specifically, it recommends that:
+1. Support for Python versions be dropped **42 months** (3.5 years) after their initial release.
```

Comment:
```suggestion
1. Support for Python versions should be continued for **42 months** (3.5 years) after their initial release.
```
I think it'd be fine for packages to support older versions of Python if they decide to (though supporting older versions of Python & dependencies sometimes comes with an increased maintenance burden).

Reply:
Thanks for this feedback Nick. I'll stick with the "drop support" language for now. I'll also add a sentence explicitly stating packages may support older Python versions if they want, which effectively means this policy has to be a "should" not a "must." I was leaning that direction anyway.